x-pack/plugin/src/test/resources/rest-api-spec/test/privileges/10_basic.yml

```diff
@@ -30,24 +30,26 @@ teardown:
         ignore: 404
 ---
 "Test put and get privileges":
-  # Single privilege, with names in URL
+  # Single privilege
   - do:
-      xpack.security.put_privilege:
-        application: app
-        name: p1
+      xpack.security.put_privileges:
         body:  >
             {
-              "application": "app",
-              "name": "p1",
-              "actions": [ "data:read/*" , "action:login" ],
-              "metadata": {
-                "key1" : "val1a",
-                "key2" : "val2a"
+              "app": {
```

Comment:
Whats the relation between `app` and `p1` and them being later declared again using `application: "app"` and `name: "p1"` do these need to be equal?

If we are taking a `Map<AppName, Map<PrivilegeName, Privilege>>` I think we should omit `application` and `name` or we have to take `Array<Privilege>`

Reply:
They are equivalent as far as I know and I agree that it is redundant. @tvernum can you chime in about this aspect?

Reply:
They're redundant and optional, but if they exist they need to match the keys from the containing object.
They're allowed for three reasons
1. Historical, originally I had the body just being an array, but there are a few problems with array bodies, including that rest tests can't handle it, so I switched to objects, and at Albert's suggestion made the inner fields optional.
2. It means that the objects that come from the GET API can be fed back into the PUT/POST without needing to remove fields
3. The format for index storage needs the fields and we re-use the parser, so we needed to do something and optional seemed better than rejection.

Reply:
Optional makes sense. I updated the rest test to use it both ways